test/topology/test_mutation_schema_change.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""
+Reproducer for a failure during lwt operation due to missing of a column mapping in schema history table.
+"""
+import asyncio
+import logging
+import time
+from test.pylib.rest_client import inject_error_one_shot, inject_error
+from test.pylib.util import wait_for
+import pytest
+from cassandra.cluster import Cluster, ConsistencyLevel  # type: ignore # pylint: disable=no-name-in-module
+from cassandra.query import SimpleStatement              # type: ignore # pylint: disable=no-name-in-module
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_mutation_schema_change(manager, random_tables):
+    """
+        Cluster A, B, C
+        create table
+        stop C
+        change schema + do lwt write + change schema
+        stop B
+        start C
+        do lwt write to the same key through C
+    """
+    server_a, server_b, server_c = await manager.running_servers()
+    t = await random_tables.add_table(ncolumns=5, pks=1)
+    manager.driver_close()
+    # Reduce the snapshot thresholds
+    await manager.mark_dirty()
+    errs = [inject_error_one_shot(manager.api, s.ip_addr, 'raft_server_snapshot_reduce_threshold')
+            for s in [server_a, server_b, server_c]]
+    await asyncio.gather(*errs)
+
+
+    logger.info("Stopping C %s", server_c)
+    await manager.server_stop_gracefully(server_c.server_id)
+    await manager.driver_connect()
+
+    async with inject_error(manager.api, server_b.ip_addr, 'paxos_error_before_learn'):
+        await t.add_column()
+        ROWS = 1
+        seeds = [t.next_seq() for _ in range(ROWS)]
+        stmt = f"INSERT INTO {t} ({','.join(c.name for c in t.columns)}) " \
+               f"VALUES ({', '.join(['%s'] * len(t.columns))}) "           \
+               f"IF NOT EXISTS"
+        query = SimpleStatement(stmt, consistency_level=ConsistencyLevel.ONE)
+        for seed in seeds:
+            logger.info("INSERT row seed %s", seed)
+            await manager.cql.run_async(query, parameters=[c.val(seed) for c in t.columns])
+        await t.add_column()
+
+    manager.driver_close()
+
+    logger.info("Stopping B %s", server_b)
+    await manager.server_stop_gracefully(server_b.server_id)
+    logger.info("Starting C %s", server_c)
+    await manager.server_start(server_c.server_id)
+
+    async def server_c_sees_another_server():
+        alive_nodes = await manager.api.get_alive_endpoints(server_c.ip_addr)
+        if len(alive_nodes) > 1:
+            return True
+    await wait_for(server_c_sees_another_server, time.time() + 45, period=.1)
+
+    logger.info("Driver connecting to C %s", server_c)
+    manager.ccluster = manager.con_gen([server_c.ip_addr], manager.port, manager.use_ssl)
```

Comment:
`[server_c.ip_addr]` is only the list of initial contact points for the driver: this is what `cluster_con` (in topology/conftest.py) does:
```python
    return Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile},
                   contact_points=hosts,
```
`hosts` is the parameter.

The driver will automatically connect to all alive nodes after querying `system.peers` from the contact points.

So there's no guarantee that the lwt write will go through C as you said in the comment. It could go through A which is also alive now.

You'd have to specify a special `load_balancing_policy` in the `ExecutionProfile` passed to `Cluster` to make sure that the write goes through C.

Reply:
I have a patch which allows you to connect to particular coordinators:

https://github.com/scylladb/scylladb/pull/13387/commits/4bea4ec1b71abc3ee9102f104c9d6d94351b42f3

Feel free to commit if good.

Reply:
@kbr-scylla @tgrabiec but what is the purpose here of going to a specific node? This is LWT, and if schema is broken on one node, we're going to have failures anyhow.  server_b is down, so LWT has to go to 2 nodes anyway. I don't see any reason to go to server_c specifically with this query.

Reply:
We test very specific code path which require going to a specific node. We are not just running an LWT query to see it if succeeds.

Reply:
I know but it doesn't require that it always goes to the same node. If it goes to a wrong node once in a while the test will succeed. Of course making the test more predictable is valuable, but even without it is good enough. That was my point. 

I tried at some point creating a connection pool that would set up a custom policy, this required very deep refactoring of the current manager code, since connections are opened on demand. 

Anyway @alecco please join the conversation and either address the comment or provide an update to the patch, I don't need to defend the patch for you I think.

Reply:
> I know but it doesn't require that it always goes to the same node. If it goes to a wrong node once in a while the test will succeed. 

Wont this just produce a fluky test?

Reply:
Query goes to the proper node now.